### PR TITLE
[#24] Bump morley ledgers

### LINKED ISSUE #24: Update morley-ledgers
# Clarification and motivation

There is an [upstream MR](https://gitlab.com/morley-framework/morley/-/merge_requests/387) that makes an important change to `ManagedLedger`. We should use it once it's merged.

# Acceptance criteria

We are using a newer version of `ManagedLedger` that stored approvals in a separate big_map.

### PR DESCRIPTION
**Pull request by rvem**
## Description
Problem: ManagedLedger implementation was updated, since our contract
reuses parts of it, we should update morley revision.

Solution: Bump morley revision, made adjustments required by updates in
the indigo and morley-ledgers.

Also, hacky lorentz calling cannot be used anymore, it makes Holdings contract very big and
impossible to originate, so now Holdings contract reuses parts of the Indigo ManagedLedger implementation.

Currently depends on https://gitlab.com/morley-framework/morley/-/merge_requests/403

## Related issue(s)

Resolves #24 

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: nettest/Main.hs [hunk 1]
```diff
@@ -5,15 +5,24 @@ module Main (main) where
...
 import Morley.Nettest
+import Util.Named
 
 import Nettest.Holdings
 import Nettest.WhitelistIntegration
 
 main :: IO ()
 main = do
-  config <- getClientConfig (Just "Globacap")
+  let clientParser = clientConfigParser . pure $ Nothing
```
Comment (gromakovsky) [inline nettest/Main.hs:15]: Why `Nothing`? It should be `Just "nettest.Globacap"` I think.

File: src/Indigo/Contracts/Holdings/Impl.hs [hunk 2]
```diff
@@ -31,11 +32,12 @@ ensureIsTransferable = do
   transferable <- getStorageField @Storage #transferable
   assertCustom_ #nonTransferable transferable
 
-holdingsContract :: ContractCode Parameter Storage
-holdingsContract = optimizeLorentz $ compileIndigoContract holdingsIndigo
+holdingsContract :: Contract Parameter Storage
+holdingsContract = defaultContract $
+  optimizeLorentz $ compileIndigoContract holdingsIndigo
```
Comment (gromakovsky) [inline src/Indigo/Contracts/Holdings/Impl.hs:40]: `defaultContract` does optimization by default.

File: src/Indigo/Contracts/Holdings/Types.hs [hunk 3]
```diff
@@ -117,14 +180,17 @@ instance TypeHasDoc (ML.StorageSkeleton StorageFields) where
 storageNotes :: Notes (ToT Storage)
 storageNotes = NTPair noAnn (ann "ledger") noAnn
```
Comment (gromakovsky) [inline src/Indigo/Contracts/Holdings/Types.hs:148]: Apparently now you can make use of https://gitlab.com/morley-framework/morley/-/merge_requests/403 here.

Comment (rvem) [reply src/Indigo/Contracts/Holdings/Types.hs:148]: oh, right, let's :zhdun: for it as well

**Comment by rvem** [comment]
old mint: 312494
old transfer: 313828
new mint: 212089
new transfer: 213296

**Comment by gromakovsky** [comment]
It should be finishable now.

**Pull request merged**